### PR TITLE
Move AtomicStore to upper level and refactor storage

### PR DESCRIPTION
**Pull request by shenkeyao**
- Moves `AtomicStore` from `AtomicKeystoreStorage` to `KeystoreSession`.
- Commits store version in the upper level.

Closes https://github.com/EspressoSystems/seahorse/issues/203.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1765,16 +1775,20 @@ impl<
...
     ) -> BoxFuture<'a, Result<Keystore<'a, Backend, L, Meta>, KeystoreError<L>>> {
-        let mut storage = AtomicKeystoreStorage::new(loader, 1024).unwrap();
+        let mut atomic_loader = AtomicStoreLoader::load(&loader.location(), "keystore").unwrap();
+        let mut storage = AtomicKeystoreStorage::new(loader, &mut atomic_loader, 1024).unwrap();
+        let mut atomic_store = AtomicStore::open(atomic_loader).unwrap();
         Box::pin(async move {
             let state = if storage.exists() {
-                storage.load().await?
+                let state = storage.load().await?;
+                state
```
Comment (jbearer) [inline src/lib.rs:1784]: Does this change do anything?

**Review by jbearer — APPROVED** [review]
Looks good, one minor little nit about a change that appears to be a noop but otherwise :+1:

File: src/lib.rs (hunk 1 above)
Comment (shenkeyao) [reply src/lib.rs:1784]: I used to add `atomic_store.commit_version()` before returning `state`, but it's no longer needed. I'll revert this change.

**Comment by shenkeyao** [comment]
@jbearer it turned out that the test failure was due to calling `Keystore::mutex.lock()` twice for `storage` and `atomic_store`. I fixed it by getting `storage` and `atomic_store` from just one `lock()` call.

**Pull request merged**